rocAL/source/decoders/image/rocjpeg_decoder.cpp

```diff
@@ -0,0 +1,386 @@
+            num_channels = 1;
+            output_image.pitch[0] = widths[0];
+            channel_sizes[0] = output_image.pitch[0] * heights[0];
+            break;
+        case ROCJPEG_OUTPUT_RGB:
+            num_channels = 1;
+            output_image.pitch[0] = widths[0] * 3;
+            channel_sizes[0] = output_image.pitch[0] * heights[0];
+            break;
+        case ROCJPEG_OUTPUT_RGB_PLANAR:
+            num_channels = 3;
+            output_image.pitch[2] = output_image.pitch[1] = output_image.pitch[0] = widths[0];
+            channel_sizes[2] = channel_sizes[1] = channel_sizes[0] = output_image.pitch[0] * heights[0];
+            break;
+        default:
+            std::cout << "Unknown output format!" << std::endl;
+            return EXIT_FAILURE;
+    }
+    return EXIT_SUCCESS;
+}
+
+/**
+ * @brief Gets the chroma subsampling string.
+ *
+ * This function gets the chroma subsampling string based on the specified subsampling value.
+ *
+ * @param subsampling The chroma subsampling value.
+ * @param chroma_sub_sampling The string to store the chroma subsampling.
+ */
+void GetChromaSubsamplingStr(RocJpegChromaSubsampling subsampling, std::string &chroma_sub_sampling) {
+    switch (subsampling) {
+        case ROCJPEG_CSS_444:
+            chroma_sub_sampling = "YUV 4:4:4";
+            break;
+        case ROCJPEG_CSS_440:
+            chroma_sub_sampling = "YUV 4:4:0";
+            break;
+        case ROCJPEG_CSS_422:
+            chroma_sub_sampling = "YUV 4:2:2";
+            break;
+        case ROCJPEG_CSS_420:
+            chroma_sub_sampling = "YUV 4:2:0";
+            break;
+        case ROCJPEG_CSS_411:
+            chroma_sub_sampling = "YUV 4:1:1";
+            break;
+        case ROCJPEG_CSS_400:
+            chroma_sub_sampling = "YUV 4:0:0";
+            break;
+        case ROCJPEG_CSS_UNKNOWN:
+            chroma_sub_sampling = "UNKNOWN";
+            break;
+        default:
+            chroma_sub_sampling = "";
+            break;
+    }
+}
+
+void HWRocJpegDecoder::initialize(int device_id, unsigned batch_size) {
+    int num_devices;
+    CHECK_HIP(hipGetDeviceCount(&num_devices));
+    if (num_devices < 1) {
+        std::cerr << "ERROR: didn't find any GPU!" << std::endl;
+        return;
+    }
+    if (device_id >= num_devices) {
+        std::cerr << "ERROR: the requested device_id is not found!" << std::endl;
+        return;
+    }
+    CHECK_HIP(hipSetDevice(device_id));
+    RocJpegBackend rocjpeg_backend = ROCJPEG_BACKEND_HARDWARE;
+    // Create stream and handle
+    CHECK_ROCJPEG(rocJpegCreate(rocjpeg_backend, device_id, &_rocjpeg_handle));
+    _rocjpeg_streams.resize(batch_size);
+    for (unsigned i = 0; i < batch_size; i++) {
+        CHECK_ROCJPEG(rocJpegStreamCreate(&_rocjpeg_streams[i]));
+    }
+
+    _device_id = device_id;
+    _batch_size = batch_size;
+    _output_images.resize(_batch_size);
+    _src_hstride.resize(_batch_size);
+    _src_img_offset.resize(_batch_size);
+
```

Comment:
Add this line here `_decode_params.resize(_batch_size);` to initialize and resize the `_decode_params` vector.

Reply:
Done